tests/Integration/BoxErrorTest.php

```diff
@@ -58,11 +58,12 @@ public function testExceptionWithErrorIsThrown() : void
             self::assertSame('AccessDeniedError', $error->getType());
             self::assertSame("Write access to space '_priv' is denied for user 'user_with_no_privileges'", $error->getMessage());
             self::assertSame(42, $error->getCode());
-            self::assertEquals([
+            /* Checks that error fields match expected. */
```

Comment:
I would prefer to keep asserting all the fields when possible. What about doing this instead:

```php
$expectedFields = [
    'object_type' => 'space',
    'object_name' => '_priv',
    'access_type' => 'Write',
    'user' => 'user_with_no_privileges',
];
if ($this->tarantoolVersionSatisfies('<{version.that.adds.user.field}')) {
    unset($expectedFields['user']);
}
self::assertEquals($expectedFields, $error->getFields());
```

Reply:
But why do we need this? AFAU the case merely tests that error raised by Tarantool is catched and reasonable error details are filled. Looks like checking depending on Tarantool version is not the purpose of this case. This will make the test break every time we make backward compatible change.

Reply:
I thought about it too, and I sort of agree with you. My main motivation for keeping all fields is that there is no dedicated unit or integration test for `getFields()` and the  [Error](https://github.com/tarantool-php/client/blob/8c45feebd39a2d41a38f28c1fea56961af24842b/src/Error.php#L16) class in general, and the correctness of the returned error object is asserted in this integration test. So I want to verify that `getFields()` returns neither more nor less than expected. And following your logic, the changes you made are questionable too, why even bother _partially_ checking the fields? We can just check that the returned field map is not empty, so the test will always be forward compatible. Between these two options I chose the first one. I agree that it could be done better, but it would require more refactoring.

Reply:
Well we can check fields of version according to `@requires Tarantool >=2.4.1` of the case annotation.

Unfortunately `$this->tarantoolVersionSatisfies('<{version.that.adds.user.field}')` will not help for CI of entrypoint version. For example for version `3.1.0-entrypoint-194-gc7dcbc406` the check boils down to:

```php
 php > echo Composer\Semver\Semver::satisfies('3.1.0-dev+entrypoint.194', '<3.1.0') ? 'true' : 'false';
false
```

Reply:
Can't we use any of this constraints? https://github.com/tarantool-php/phpunit-extras/blob/bd460c94c079c8809ee7653faed1237922c797ea/tests/Annotation/Requirement/TarantoolVersionRequirementTest.php#L87-L94

Reply:
Sorry, `false` is correct for the case when dev version has `user` field (after my PR). But it is not in the other case.

So it looks inconvenient. See, if we merge this PR then Tarantool PR's start to fail until PR with 'user' field is merged. I hope to merge mine soon yet. It is a matter of passing CI.

Reply:
Make a version depended check as discussed f2f.

<details>
<summary>diff</summary>

```diff
@@ -58,11 +58,16 @@ final class BoxErrorTest extends TestCase
             self::assertSame('AccessDeniedError', $error->getType());
             self::assertSame("Write access to space '_priv' is denied for user 'user>
             self::assertSame(42, $error->getCode());
-            self::assertEquals([
+            $expectedFields = [
                 'object_type' => 'space',
                 'object_name' => '_priv',
                 'access_type' => 'Write',
-            ], $error->getFields());
+                'user' => 'user_with_no_privileges',
+            ];
+            if ($this->tarantoolVersionSatisfies('<= 3.0')) {
+                unset($expectedFields['user']);
+            }
+            self::assertEquals($expectedFields, $error->getFields());
             self::assertNull($error->getPrevious());
         }
     }
```

</details>